src/sentry/replays/usecases/ingest.py

```diff
@@ -248,3 +218,54 @@ def process_headers(bytes_with_headers: bytes) -> tuple[RecordingSegmentHeaders,
 
 def replay_recording_segment_cache_id(project_id: int, replay_id: str, segment_id: str) -> str:
     return f"{project_id}:{replay_id}:{segment_id}"
+
+
+def store_replays_with_filestore(
+    message: RecordingIngestMessage, headers: RecordingSegmentHeaders, recording_segment: bytes
+) -> None:
+    recording_segment_file_name = f"rr:{message.replay_id}:{headers['segment_id']}"
+    with metrics.timer("replays.store_recording.store_recording.create_file"):
+        file = File.objects.create(
+            name=recording_segment_file_name,
+            type="replay.recording",
+        )
+    with metrics.timer("replays.store_recording.store_recording.put_segment_file"):
+        file.putfile(
+            BytesIO(recording_segment),
+            blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE,
+        )
+
+    try:
+        # associate this file with an indexable replay_id via ReplayRecordingSegmentModel
+        with metrics.timer("replays.store_recording.store_recording.create_segment_row"):
+            ReplayRecordingSegmentModel.objects.create(
+                replay_id=message.replay_id,
+                project_id=message.project_id,
+                segment_id=headers["segment_id"],
+                file_id=file.id,
+                size=len(recording_segment),
+            )
+    except IntegrityError:
+        # Raised in the event of a concurrent write.  Reasonably space
+        logger.warning(
+            "Recording-segment has already been processed.",
+            extra={
+                "replay_id": message.replay_id,
+                "project_id": message.project_id,
+                "segment_id": headers["segment_id"],
+            },
+        )
+
+        # Cleanup the blob.
+        file.delete()
+
+
+def store_replays_directly(
+    message: RecordingIngestMessage, headers: RecordingSegmentHeaders, recording_segment: bytes
+) -> None:
+    storage = get_storage(replays_storage_options())
+    file_path = f"{message.project_id}/{message.replay_id}/{headers['segment_id']}"
+    storage.save(
```

Comment:
our File model will create 1 file per 1 MB. this currently will not do that. Given our small amount of files above 1 MB and our cap, this should be fine for now.

Reply:
also kind of interesting that we have a 1MB kafka limit, buffer all of the file chunks in redis, then write them to our file storage in 1MB chunks. makes you wonder why we don't just write the chunks directly :wink: